openassessment/xblock/static/js/src/lms/oa_staff_area.js

```diff
@@ -115,6 +118,54 @@
             return deferred.promise();
         },
 
+        loadStaffGradeForm: function(eventObject) {
+            var view = this;
+            var staff_form_element = $(eventObject.currentTarget);
+            var isCollapsed = staff_form_element.hasClass("is--collapsed");
+            var deferred = $.Deferred();
+
+            if (isCollapsed && !this.staffGradeFormLoaded) {
+                eventObject.preventDefault();
+                this.staffGradeFormLoaded = true;
+                this.server.staffGradeForm().done(function(html) {
+                    //// Clear any error message
+                    //showFormError('');
+
+                    // Load the HTML and install event handlers
+                    $('.staff__grade__form', view.element).replaceWith(html);
+
+                    // Initialize the rubric : TODO SHARE CODE!
+                    var $rubric = $('.staff-assessment__assessment', view.element);
+                    if ($rubric.size() > 0) {
+                        var rubricElement = $rubric.get(0);
+                        var rubric = new OpenAssessment.Rubric(rubricElement);
+
+                        // Install a change handler for rubric options to enable/disable the submit button
+                        rubric.canSubmitCallback($.proxy(view.staffSubmitEnabled, view));
```

Comment:
Note that there are some interactions with the submit button up in the student-facing portion of the page (when a peer assessment view is open). We will need to do better scoping.